### PR TITLE
Feature01 jest test for ResClient  in Rest-hook package

### PR DESCRIPTION
**Pull request by chjuruena**
# ResClient Jest TEst

## Description
This PR introduces Jest tests for the `createRestClient` and `_expandRestPath` functions in the project. These tests ensure the correctness and reliability of the REST client creation and path expansion functionalities.

## Changes Proposed
- Added Jest package (`jest`) and its related typings (`@types/jest`) to the project dependencies.
- Implemented Jest tests for the `createRestClient` and `_expandRestPath` functions to enhance test coverage and ensure code quality.
- Updated the project's testing script to use `pnpm test` to run Jest tests.

## Testing

- Run `pnpm test` to run jest tests

#### `createRestClient` Tests
- [ ] Added test to ensure the REST client is created with provided configurations

#### `_expandRestPath` Tests
- [ ] Added test to verify REST path expansion with given parameters

## Checklist
- [x] I have reviewed my own code
- [x] I have performed a self-review of my changes
- [ ] I have added/updated relevant documentation (if necessary)
- [ ] All tests pass locally
- [ ] Any dependent changes have been merged and published in downstream modules (if applicable)

### CODE REVIEW

File: packages/rest-hooks/jest.config.ts [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+  clearMocks: true,
+
+  // Indicates whether the coverage information should be collected while executing the test
+  collectCoverage: true,
+
+  // An array of glob patterns indicating a set of files for which coverage information should be collected
+  // collectCoverageFrom: undefined,
+
+  // The directory where Jest should output its coverage files
+  coverageDirectory: 'coverage',
```
Comment (Bexanderthebex) [inline packages/rest-hooks/jest.config.ts:28]: Should also have a `.gitignore` that excludes this folder from being committed to git

File: packages/rest-hooks/jest.config.ts [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+  // testEnvironmentOptions: {},
+
+  // Adds a location field to test results
+  // testLocationInResults: false,
+
+  // The glob patterns Jest uses to detect test files
+  testMatch: [
+    '<rootDir>/__tests__/**/*.[jt]s?(x)',
+    '**/?(*.)+(spec|test).[tj]s?(x)',
+    '<rootDir>/src/**/*.test.[jt]s?(x)'
```
Comment (Bexanderthebex) [inline packages/rest-hooks/jest.config.ts:161]: Any reason why we have the two other entries besides `'**/?(*.)+(spec|test).[tj]s?(x)'`?

File: packages/rest-hooks/jest.config.ts [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+  // watchman: true,
+  'moduleFileExtensions': [
+    'js',
+    'jsx',
+    'ts',
+    'tsx'
+  ],
+  'preset': 'ts-jest'
+
+
```
Comment (Bexanderthebex) [inline packages/rest-hooks/jest.config.ts:207]: remove new line

File: packages/rest-hooks/package.json [hunk 4]
```diff
@@ -26,11 +30,13 @@
     "axios": "^1.6.7",
     "json-stringify-safe": "^5.0.1",
     "ramda": "^0.29.1",
-    "ramda-adjunct": "^4.1.1"
+    "ramda-adjunct": "^4.1.1",
+    "ts-node": "^10.9.2"
   },
   "scripts": {
     "build": "pnpm run clean && tsup && ls dist",
```
Comment (Bexanderthebex) [inline packages/rest-hooks/package.json:37]: Please follow up with steve on this. It's been 2 weeks

File: packages/rest-hooks/src/__tests__/restClient.test.ts [hunk 5]
```diff
@@ -0,0 +1,65 @@
+import { jest } from '@jest/globals' // Import jest from @jest/globals
+import axios from 'axios'
+import { createRestClient, _expandRestPath } from '../restClient'
+
+// Mock axios.create to prevent actual network requests
```
Comment (Bexanderthebex) [inline packages/rest-hooks/src/__tests__/restClient.test.ts:5]: So what is the connection of axios to the module of our testing? Comments should add that context but I think that the developer could easily gather that information once he comes in and reads the module code. So if it was me, I would remove it. But I will leave it to you how you will tackle this.

File: packages/rest-hooks/src/__tests__/restClient.test.ts [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+
+    expect(restClient.clientConfig).toEqual(clientConfig)
+    expect(restClient.serverConfig).toEqual(serverConfig)
+    expect(axios.create).toHaveBeenCalledWith({
+      baseURL: serverConfig.defaultBaseUrl,
+      timeout: serverConfig.timeout,
+    })
+  })
+
+  // Add more tests as needed
```
Comment (Bexanderthebex) [inline packages/rest-hooks/src/__tests__/restClient.test.ts:36]: How about the HTTP verb calls?

File: packages/rest-hooks/src/__tests__/restClient.test.ts [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+      queryParams: { hydrate: true }
+    }
+    const expectedOutput = '/v1/spark/appplatforms/22?hydrate=true'
+
+    const result = _expandRestPath(restPath, params)
+
+    expect(result).toEqual(expectedOutput)
+  })
+
+  // Add more tests as needed
```
Comment (Bexanderthebex) [inline packages/rest-hooks/src/__tests__/restClient.test.ts:64]: It's a default that we will need to be adding test to we can remove this

File: packages/rest-hooks/jest.config.ts (hunk 1 above)
Comment (chjuruena) [reply packages/rest-hooks/jest.config.ts:28]: Added them in https://github.com/SteveAtSentosa/stcland/pull/4/files#diff-bc37d034bad564583790a46f19d807abfe519c5671395fd494d8cce506c42947R28

File: packages/rest-hooks/src/__tests__/restClient.test.ts (hunk 7 above)
Comment (chjuruena) [reply packages/rest-hooks/src/__tests__/restClient.test.ts]: Removed this already

File: packages/rest-hooks/src/__tests__/restClient.test.ts (hunk 5 above)
Comment (chjuruena) [reply packages/rest-hooks/src/__tests__/restClient.test.ts]: Remove this also

File: packages/rest-hooks/jest.config.ts (hunk 2 above)
Comment (chjuruena) [reply packages/rest-hooks/jest.config.ts]: I removed the duplicate, i was not able to remove this after correctly installing ts-node

File: packages/rest-hooks/src/__tests__/restClient.test.ts (hunk 6 above)
Comment (chjuruena) [reply packages/rest-hooks/src/__tests__/restClient.test.ts]: Added them in my latest chanes

**Review by Bexanderthebex — APPROVED** [review]

**Comment by Bexanderthebex** [comment]
LGTM to me, feel free to merge

**Pull request merged**